docs/content/modeling/migrating-relations.mdx

```diff
@@ -0,0 +1,310 @@
+- [Intersection Operator](../configuration-language.mdx#the-intersection-operator): the intersection operator can be used to indicate a relationship exists if the user is in all the sets of users
+
+</details>
+
+## Step By Step
+
+Assume that you want to migrate to the following model:
+
+<AuthzModelSnippetViewer
+  configuration={{
+    type_definitions: [
+      {
+        type: 'document',
+        relations: {
+          writer: {
+            this: {},
+          },
+          can_write: {
+            computedUserset: {
+              object: '',
+              relation: 'writer',
+            },
+          },
+        },
+      },
+      {
+        type: 'user',
+      },
+    ],
+  }}
+/>
+
+### 01. Creating a Backward Compatible Model
+
+To create a seamless transition, we want to create a backward compatible model so that existing tuple will still work.
+
+<AuthzModelSnippetViewer
+  configuration={{
+    type_definitions: [
+      {
+        type: 'document',
+        relations: {
+          editor: {
+            this: {},
+          },
+          writer: {
+            union: {
+              child: [
+                {
+                  this: {},
+                },
+                {
+                  computedUserset: {
+                    relation: 'editor',
+                  },
+                },
+              ],
+            },
+          },
+          can_write: {
+            computedUserset: {
+              object: '',
+              relation: 'writer',
+            },
+          },
+          can_edit: {
+            computedUserset: {
+              object: '',
+              relation: 'writer',
+            },
+          },
+        },
+      },
+      {
+        type: 'user',
+      },
+    ],
+  }}
+/>
+
+For the above model, check on whether user:Anne can write document:roadmap will produce the correct value while can_edit will still work.
+
+<CheckRequestViewer user={'user:anne'} relation={'can_write'} object={'document:roadmap'} allowed={true} />
+<CheckRequestViewer user={'user:anne'} relation={'can_edit'} object={'document:roadmap'} allowed={true} />
+
+### 02. New Relationship Tuples Should Use New Relation
+
+After we have updated to the backward compatible models, new relationship tuples should use the new relation.
+
+For example, we will want to add Bethany to writer instead of editor relationship.
+
+<WriteRequestViewer
+  relationshipTuples={[
+    {
+      _description: 'Bethany is to be assigned as writer instead of editor',
+      user: 'user:bethany',
+      relation: 'writer',
+      object: 'document:roadmap',
+    },
+  ]}
+/>
+
+We can verify by calling check API for Bethany
+<CheckRequestViewer user={'user:bethany'} relation={'can_write'} object={'document:roadmap'} allowed={true} />
```

Comment:
```suggestion
## Create a new relationship tuple

Now that you have a backwards compatible model, you can create new relationship tuples with a new relation.

In this example, you will add Bethany to the `writer` relationship.

<WriteRequestViewer
  relationshipTuples={[
    {
      _description: 'Bethany assigned writer instead of editor',
      user: 'user:bethany',
      relation: 'writer',
      object: 'document:roadmap',
    },
  ]}
/>

Run a check in the API for Bethany to ensure correct access.

<CheckRequestViewer user={'user:bethany'} relation={'can_write'} object={'document:roadmap'} allowed={true} />
```